packages/fork-choice/src/forkChoice/forkChoice.ts

```diff
@@ -162,25 +173,35 @@ export class ForkChoice implements IForkChoice {
 
     let timer;
     this.metrics?.forkChoiceRequests.inc();
-
     try {
+      let deltas: number[];
+
+      // Check if scores need to be calculated/updated
       if (!this.synced) {
         timer = this.metrics?.forkChoiceFindHead.startTimer();
-        const deltas = computeDeltas(
-          this.protoArray.indices,
-          this.votes,
-          this.justifiedBalances,
-          this.justifiedBalances
-        );
+        deltas = computeDeltas(this.protoArray.indices, this.votes, this.justifiedBalances, this.justifiedBalances);
+        /**
+         * The structure in line with deltas to propogate boost up the branch
+         * starting from the proposerIndex
+         */
+        const boosts: number[] = [];
+        if (this.proposerBoost) {
```

Comment:
if we call multiple `updateHead()` (with some `onAttestation()`) in the middle, we should not apply boost multiple times, please add this case in a unit test if you can

Reply:
computeDeltas always use `balanceWeight` of the node (which is the original weight without proposer boost) and  adds boost to calculate the final `weight`, so this will not add `boost` again.

Reply:
test added here: https://github.com/ChainSafe/lodestar/pull/3540/files#diff-7153f5b72303272f1458508e23986e3ed13dc46ad1347903db75dca5f8821d7fR86 :+1: